### ISSUE TITLE
Console Elements flickering on interaction

### ISSUE THREAD

**Issue by Paffo** [migueldeicaza/gui.cs#1353]
Hi,
Using the Dotnet Console (and Powershell) while navigating with the mouse and selecting checkboxes or focusing textfields,
the rest of the Gui disappares.
I have no idea whats causing the issue or what im doing wrong.
my Application.Top has a TopLevel and a Menu.
The TopLevel has a TabView and a Label
The TabView (every tab) has a Frame
All the Elements (Textfields, labels, checkboxes) are inside this Frames.

Tried removing all the logic leaving only graphics but it still does it.

![issue](https://user-images.githubusercontent.com/15388318/123336838-b2da2580-d546-11eb-89f0-2edecbaa68c7.gif)

**Comment by tznind** [comment]
> [...] while navigating with the mouse and selecting checkboxes or focusing textfields,
> the rest of the Gui disappares.

Is it only mouse navigation that causes this issue? Or does keyboard navigation also have issues?

**Comment by tznind** [comment]
Ok so if I change the code to this:

```csharp
Application.Init ();

			var win = new Window () { Width = Dim.Fill (), Height = Dim.Fill() };

			var tabView = new TabView () { Width = Dim.Fill (), Height = Dim.Fill () };
			var frame1 = new FrameView ("Frame1") { Width = Dim.Fill (), Height = Dim.Fill () };
			var frame2 = new FrameView ("Frame2") { Width = Dim.Fill (), Height = Dim.Fill () };

			frame1.Add (new Label { Text = "Hi", Y = 0 });
			frame1.Add (new CheckBox ("Option1") { Y = 1 });
			frame1.Add (new CheckBox ("Option2") { Y = 2 });

			tabView.AddTab (new TabView.Tab ("Tab1", frame1), false);
			tabView.AddTab (new TabView.Tab ("Tab2", frame2), false);
			win.Add (tabView);

			Application.Top.Add (win);
			Application.Run ();
			Application.Shutdown ();
```

Then the problem disapears.

**Comment by BDisp** [comment]
Yes, you right. But if you add the follow code to the end of the `FrameView` `Redraw` you will see that works with your original code:

```cs
// Checks if there are any SuperView view which intersect with this window.
if (SuperView != null) {
	SuperView.SetNeedsLayout ();
	SuperView.SetNeedsDisplay ();
}

```

**Comment by BDisp** [comment]
If you want, feel free to submit a PR for this.

**Comment by tznind** [comment]
That sounds like a simple fix.  Can you create the PR? I'm not sure I understand how/why the fix works so wouldn't feel comfortable creating it.

**Comment by BDisp** [comment]
@tznind, I thought better but the `TabView.Tab` has any call to the `SetNeedsDisplay` to `SuperView` anytime a subview was changes?

**Comment by tznind** [comment]
So `TabView` has 2 private field `Views`.  The first is a `TabRowView` (called tabsBar) and the second is a regular `View` (called contentView).  Neither of these views are child views of the main `TabView` control (i.e. they are not added using `Add(tabsBar)`) But are instead drawn manually as part of the `TabView` Redraw implementation:

https://github.com/migueldeicaza/gui.cs/blob/8f66fc07c79c2841a4a9397eaa22796e3719173d/Terminal.Gui/Views/TabView.cs#L165-L184

Maybe this causes problems since from the point of view of the checkbox they aren't going to appear as part of the view hierarchy?

I will have an investigate to confirm this and see if there is a better solution than the current implementation or a workaround.

**Comment by BDisp** [comment]
@tznind, I think is better you do the changes because it concerns to fixing the `Redraw` method in the `TabView`.

```cs
if (Tabs.Any ()) {
	tabsBar.Redraw (tabsBar.Bounds);
	contentView.SetNeedsDisplay ();
	contentView.Redraw (contentView.Bounds);
}

```

**Comment by tznind** [comment]
Ignore my deleted post there, I am talking rubbish.  The subviews of `TabView` are indeed part of the normal hierarchy and are added in the constructor

**Comment by tznind** [comment]
Thanks for the help finding that fix, I don't think I would have ever found that myself.  

@Paffo there is a PR in to fix this.  In the meantime it might be that you can do a workaround based on one of my above examples?

**Comment by tig** [comment]
Fixed in #1345

**Comment by BDisp** [comment]
> 
> 
> Fixed in #1345

@tig please correct the fix for this issue. It's hte PR #1354. Thanks.
